packages/hardhat-core/sample-projects/javascript-esm/test/Lock.js

```diff
@@ -7,22 +7,25 @@ import chai from "chai";
 
 const { expect } = chai;
 
+import LockModule from "../ignition/modules/LockModule";
+
 describe("Lock", function () {
   // We define a fixture to reuse the same setup in every test.
   // We use loadFixture to run this setup once, snapshot that state,
   // and reset Hardhat Network to that snapshot in every test.
   async function deployOneYearLockFixture() {
     const ONE_YEAR_IN_SECS = 365 * 24 * 60 * 60;
-    const ONE_GWEI = 1_000_000_000;
+    const ONE_GWEI = 1_000_000_000n;
 
     const lockedAmount = ONE_GWEI;
     const unlockTime = (await time.latest()) + ONE_YEAR_IN_SECS;
 
     // Contracts are deployed using the first signer/account by default
     const [owner, otherAccount] = await ethers.getSigners();
 
-    const Lock = await ethers.getContractFactory("Lock");
-    const lock = await Lock.deploy(unlockTime, { value: lockedAmount });
+    const { lock } = await ignition.deploy(LockModule, {
```

Comment:
I think we should be explicit and import `ignition` or the `hre`, even if it works without it. The TS sample projects seem to be doing the latter. Same applies to the javascript tests.

Reply:
Correction: one of the TypeScript sample projects uses the hre directly, while another destructures it. We should ensure consistency across all sample projects, regardless of the approach we choose.

Reply:
The approach used so far has been: js implicit, ts explicit.

 But regarding `hre.` vs destructuring: that's less clear.

Reply:
Implicit sounds a bit like magic. But if that's a conscious decision I'm good with it. Regarding destructuring vs using hre, I think the value of using hre is that the code is more explicit. For example, you see a deployContract in the code, is that coming from hardhat, viem or ethers? If we recommend doing hre.viem.deploy contract the origin is crystal clear.

Reply:
I have went for `hre.ethers` and `hre.viem`, I think it is good to be explicit as we enhance viem/ethers, the `hre` prefix makes it clear that you should be looking to the HH plugin for docs on the method.

Reply:
I opted to go for `hre.ethers` and `hre.viem` in the typescript based projects. I have left the js ones implicit.
I think that given the viem/ethers objects are enhanced by HH in the plugins, we should be explicit; `ethers` should be `hre.ethers` if it is coming from us so that `ethers` can mean imported from the ethers library.